assets/js/utils/dom.ts

```diff
@@ -0,0 +1,93 @@
+// DOM Utils
+
+/**
+ * Get the first matching element
+ */
+export function $<E extends Element = Element>(selector: string, context: Pick<Document, 'querySelector'> = document): E | null {
+  return context.querySelector<E>(selector);
+}
+
+/**
+ * Get every matching element as an array
+ */
+export function $$<E extends Element = Element>(selector: string, context: Pick<Document, 'querySelectorAll'> = document): E[] {
+  const elements = context.querySelectorAll<E>(selector);
+
+  return [...elements];
+}
+
+export function showEl<E extends HTMLElement>(...elements: E[] | ConcatArray<E>[]) {
+  ([] as E[]).concat(...elements).forEach(el => el.classList.remove('hidden'));
+}
+
+export function hideEl<E extends HTMLElement>(...elements: E[] | ConcatArray<E>[]) {
+  ([] as E[]).concat(...elements).forEach(el => el.classList.add('hidden'));
+}
+
+export function toggleEl<E extends HTMLElement>(...elements: E[] | ConcatArray<E>[]) {
+  ([] as E[]).concat(...elements).forEach(el => el.classList.toggle('hidden'));
+}
+
+export function clearEl<E extends HTMLElement>(...elements: E[] | ConcatArray<E>[]) {
+  ([] as E[]).concat(...elements).forEach(el => {
+    while (el.firstChild) el.removeChild(el.firstChild);
+  });
+}
+
+export function removeEl<E extends HTMLElement>(...elements: E[] | ConcatArray<E>[]) {
+  ([] as E[]).concat(...elements).forEach(el => el.parentNode?.removeChild(el));
+}
+
+export function makeEl<Tag extends keyof HTMLElementTagNameMap>(tag: Tag, attr?: Partial<HTMLElementTagNameMap[Tag]>): HTMLElementTagNameMap[Tag] {
+  const el = document.createElement(tag);
+  if (attr) {
+    for (const prop in attr) {
+      const newValue = attr[prop];
+      if (typeof newValue !== 'undefined') {
+        el[prop] = newValue as Exclude<typeof newValue, undefined>;
+      }
+    }
+  }
+  return el;
+}
+
+export function insertBefore(existingElement: HTMLElement, newElement: HTMLElement) {
+  existingElement.parentNode?.insertBefore(newElement, existingElement);
```

Comment:
Isn't this just insertAdjacentElement? Also, I checked in the source code and unless I'm missing something this function doesn't actually appear to be used anywhere, so can be deleted.